tests/test_parse_shim.py

```diff
@@ -0,0 +1,101 @@
+from dataclasses import dataclass
+
+import pytest
+
+from bleach.parse_shim import urlparse
+
+
+@dataclass
+class ParseResult:
+    scheme: str = ""
+    netloc: str = ""
+    path: str = ""
+    params: str = ""
+    query: str = ""
+    fragment: str = ""
```

Comment:
Nice! I think this makes the table-based tests cleaner.

Possible future TODO would be to capture errors in this class like a Rust Result, but I guess Python usually avoids that?

Reply:
Oh, that's a good idea. I'll add a failure test and see how that might work.